toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,236 @@
-			    popup.add(stopRecordingItem);
-			    popup.addSeparator();
-			    popup.add(killItem);
-			    popup.addSeparator();
-			    popup.add(settingsItem); 
-			    
-			    this.trayIcon = new TrayIcon(offline_image, "Toast TK - Web Agent", popup);
-			} catch (IOException e1) {
-				LOG.info(e1);
+			    this.trayIcon = new TrayIcon(offlineImage, "Toast TK - Web Agent", popup);
+				 trayIcon.setImageAutoSize(true);
+				 tray.add(trayIcon);
+			} catch (IOException|AWTException e1) {
+				LOG.error(e1);
 			}
-		    trayIcon.setImageAutoSize(true);
-		    try {
-		        tray.add(trayIcon);
-		    } catch (AWTException e) {
-		    	LOG.info(e);
-		    }
 		}
 	}
 	
-	ActionListener getConnectListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	try {
-					if(verificationWebApp(webAppName)) {
-						if(agentServer.register(webConfigProvider.get().getApiKey())) {
-							trayIcon.setImage(online_image);
-							connectedToWebApp = true;
-							NotificationManager.showMessage("Web Agent - Connected to Webapp !").showNotification();
-						}
-						else
-							NotificationManager.showMessage("The ApiKey does not match with the WebApp: \n" + webConfigProvider.get().getApiKey()).showNotification();
-					}
-					else 
-						NotificationManager.showMessage("The Web App does not anwser").showNotification();
-				} catch (IOException e1) {
-					LOG.error(e1.getMessage(), e1);
-				}
-	        }
-	    };
-	    return listener;
+	private PopupMenu initMenuItem() {
+		PopupMenu popup = new PopupMenu();
+	    
+	    MenuItem quitItem = new MenuItem("Quit");
+	    MenuItem connectItem = new MenuItem("Connect");
+	    MenuItem executeItem = new MenuItem("Execute Scripts");
+	    MenuItem startRecordingItem = new MenuItem("Start Recording");
+	    MenuItem stopRecordingItem = new MenuItem("Stop Recording");
+	    MenuItem settingsItem = new MenuItem("Settings");
+	    
+	    quitItem.addActionListener(this::killListener);
+	    connectItem.addActionListener(this::connectListener);
+	    executeItem.addActionListener(this::executeListener);
+	    stopRecordingItem.addActionListener(this::stopListener);
+	    startRecordingItem.addActionListener(this::start);
+	    settingsItem.addActionListener(this::settingsListener);
+	    
+	    popup.add(connectItem);
+	    popup.addSeparator();
+	    popup.add(startRecordingItem);
+	    popup.add(stopRecordingItem);
+	    popup.addSeparator();
+	    popup.add(executeItem);
+	    popup.addSeparator();
+	    popup.add(settingsItem);
+	    popup.addSeparator();
+	    popup.add(quitItem); 
+	    
+	    return popup;
 	}
 	
-	ActionListener getKillListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	agentServer.unRegister();
-	        	System.exit(-1);
-	        }
-	    };
-	    return listener;
+	private void connectListener(ActionEvent e){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "e". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)